packages/components/src/public/QueryModel/CustomizeGridViewModal.tsx

```diff
@@ -150,20 +200,27 @@ export const CustomizeGridViewModal: FC<Props> = memo(props => {
                             {/* Taking this out for now, until we figure out how to handle session views here */}
                             {/*{!model.currentView.session && isDirty && <span className="pull-right action-text" onClick={revertEdits}>Restore default columns</span>}*/}
                         </div>
-                        <div className="list-group field-modal__col-content">
-                            {
-                                columnsInView.map((column, index) => {
-                                    return (
-                                        <ColumnInView
-                                            key={index}
-                                            column={column}
-                                            index={index}
-                                            onColumnRemove={() => removeColumn(index)}
-                                        />
-                                    )
-                                })
-                            }
-                        </div>
+                        <DragDropContext onDragEnd={onDropField} >
+                            <Droppable droppableId="field-droppable">
+                                {dropProvided => (
+                                    <div className="list-group field-modal__col-content" {...dropProvided.droppableProps} ref={dropProvided.innerRef}>
+                                        {columnsInView.map((column, index) => {
+                                            return (
+                                                <ColumnInView
+                                                    key={index}
+                                                    column={column}
+                                                    index={index}
+                                                    onColumnRemove={() => removeColumn(index)}
+                                                    selected={selectedIndex === index}
+                                                    onClick={() => onSelectField(index)}
```

Comment:
I believe I fixed the `onColumnRemove` one in my FB, but we should refactor so that we aren't using "arrow functions as props" (see IntelliJ warning).